.github/workflows/frontend_charm_integration_tests.yml

```diff
@@ -1,45 +1,26 @@
 name: Run charm tests
 on:
   push:
-    branches: ["main"]
+    branches: ["main", "integration-test-fix*"]
     tags: ["v*.*.*"]
 
 jobs:
   integration-test:
+    permissions:
+      contents: read
+      packages: read
+
     name: Integration tests
-    runs-on: [self-hosted, linux, large]
+    runs-on: [self-hosted, jammy, xlarge]
     steps:
       - name: Checkout
-        uses: actions/checkout@v3
-      - name: Setup LXD
-        uses: canonical/setup-lxd@main
-        with:
-          channel: 5.13/stable
-      - name: Setup operator environment
-        uses: charmed-kubernetes/actions-operator@main
-        with:
-          provider: microk8s
+        uses: actions/checkout@v4
+      - name: Set up microk8s
+        uses: canonical/certification-github-workflows/.github/actions/microk8s-setup@microk8s-setup-action
```

Comment:
The setup steps turned out to be more extensive on self hosted runners, and since this is all repetitive boilerplate, moved to a reusable composite action. Same story below re: the `archive-charm-testing-artifacts`